backend/src/main/kotlin/com/group7/artshare/controller/ArtItemController.kt

```diff
@@ -23,12 +27,20 @@ class ArtItemController(
     @Autowired
     lateinit var artItemRepository: ArtItemRepository
 
+    @Autowired
+    lateinit var artistRepository: ArtistRepository
+
+    @Autowired
+    lateinit var registeredUserRepository: RegisteredUserRepository
+
     @GetMapping("{id}")
-    fun getRecommendedArtItemsGeneric(@PathVariable("id") id: Long): ArtItem =
-        artItemRepository.findByIdOrNull(id) ?: throw ResponseStatusException(
-            HttpStatus.NOT_FOUND,
-            "Id is not match with any of the art items in the database"
-        )
+    fun getRecommendedArtItemGeneric(@PathVariable("id") id: Long) : ArtItemDTO? {
```

Comment:
We can remove nullable return type by changing the code like this
I recommend not using methods like Objects.nonNull. It is better to use Kotlin way of null checking because if you do it other way, compiler may not understand that you checked. So it gives warning that says it can be null make the return type nullable etc.